web/src/test/java/com/microsoft/applicationinsights/web/utils/JettyTestServer.java

```diff
@@ -26,14 +26,40 @@
 import org.eclipse.jetty.servlet.FilterMapping;
 import org.eclipse.jetty.servlet.ServletContextHandler;
 
+import java.io.IOException;
+import java.net.ServerSocket;
+
 /**
  * Created by yonisha on 2/3/2015.
  */
 public class JettyTestServer {
     private Server server;
-    private int Min = 1050;
-    private int Max = 15000;
-    private int portNumber = Min + (int)(Math.random() * ((Max - Min) + 1));
+    private final int Min = 1050;
+    private final int Max = 15000;
+    private final int portNumber;
+
+    public JettyTestServer() {
+        // try 256 times for ports
+        int initialPortNumber = Min + (int)(Math.random() * ((Max - Min) + 1));
+        for (int i = 0; i < 256; i++) {
+            ServerSocket ss = null;
+            try {
+                ss = new ServerSocket(initialPortNumber);
```

Comment:
@littleaj I am bit confused! I thought the purpose was to try different ports until we find an open port. If this is true I find 2 concerns:
1. the `initialPortNumber` is created outside for loop and I don't see where it is getting updated in the loop.
2. The loop never breaks after finding the successful port thus always running for 256 times. 
Either, I am fully lost or may be you need to update this.

Reply:
1. I don't want to generate a random number each time because it could choose the same one twice.
2. lol...yes, it does need a `break`.